src/ReinforcementLearningEnvironments/src/environments/3rd_party/gym.jl

```diff
@@ -69,26 +69,45 @@ RLBase.action_space(env::GymEnv) = env.action_space
 RLBase.state_space(env::GymEnv) = env.observation_space
 
 function RLBase.reward(env::GymEnv{T}) where {T}
-    if pyisinstance(env.state, PyCall.@pyglobalobj :PyTuple_Type) && length(env.state) == 4
-        obs, reward, isdone, info = convert(Tuple{T,Float64,Bool,PyDict}, env.state)
+    if pyisinstance(env.state, PyCall.@pyglobalobj :PyTuple_Type) && length(env.state) == 5
+        _, reward, = convert(Tuple{T,Float64,Bool,Bool,PyDict}, env.state)
+        reward
+    elseif pyisinstance(env.state, PyCall.@pyglobalobj :PyTuple_Type) && length(env.state) == 4
+        _, reward, = convert(Tuple{T,Float64,Bool,PyDict}, env.state)
         reward
     else
         0.0
     end
 end
 
 function RLBase.is_terminated(env::GymEnv{T}) where {T}
-    if pyisinstance(env.state, PyCall.@pyglobalobj :PyTuple_Type) && length(env.state) == 4
-        obs, reward, isdone, info = convert(Tuple{T,Float64,Bool,PyDict}, env.state)
+    if pyisinstance(env.state, PyCall.@pyglobalobj :PyTuple_Type) && length(env.state) == 5
+        _, _, isterminated, = convert(Tuple{T,Float64,Bool,Bool,PyDict}, env.state)
+        isterminated
+    elseif pyisinstance(env.state, PyCall.@pyglobalobj :PyTuple_Type) && length(env.state) == 4
+        @warn "Gym version outdated. Update gym to obtain termination and truncation info instead of done signal."
+        _, _, isdone, = convert(Tuple{T,Float64,Bool,PyDict}, env.state)
         isdone
     else
         false
     end
 end
 
+function is_truncated(env::GymEnv{T}) where {T}
```

Comment:
What is this function for?

Reply:
By now gym differentiates between truncated and (naturally) terminated episodes. In some cases it makes sense to know the difference because you might want to bootstrap the future value in case of a truncated episode. In contrast, if an episode naturally terminates (e.g. after reaching a goal), there is no future value. 
I thought it would make sense to incorporate this functionality in gym.jl.

Reply:
Sure, good idea 👍

Reply:
I just commited this change. Now whenever the truncated **or** terminated flag is true, is_terminated will return true.